### ISSUE TITLE
merge our projects

### ISSUE THREAD

**Issue by oli-obk** [pnkfelix/cmd-pandoc#1]
I have [also implemented](https://github.com/oli-obk/rust-pandoc) a pandoc wrapper. We could merge our efforts.

**Comment by pnkfelix** [comment]
Yeah I actually tried to port [tango](https://github.com/pnkfelix/tango) to use your `pandoc` wrapper, but there was some problem that stopped me from doing so (I think the first issue was that there was some command line parameter your `pandoc` didn't support. And then the other issue was that I think mine might have been providing some high-level procedure that happened to be useful for `tango`; but with the right abstraction it should be reasonable to merge our efforts, and have `tango` use the merged product.

I suspect the best way to go here, since you clearly have reserved the superior name for this crate, is that I should open up issues and PR's against your `pandoc` repo.  How does that sound?

**Comment by oli-obk** [comment]
> I suspect the best way to go here, since you clearly have reserved the superior name for this crate, is that I should open up issues and PR's against your pandoc repo. How does that sound?

Superb. Heads-up so we don't duplicate effort: I'm going to take your command line arg enum and merge it into my crate.

**Comment by oli-obk** [comment]
Oh one more thing: I have _no_ clue about licensing. Can I just rip all the licensing comments/files you made here and copy it to my crate?

**Comment by pnkfelix** [comment]
@oli-obk are you asking if you can just copy over the licenses as is into your crate?

Or are you asking if you can _rip out_ the licenses and copy in mine into your crate where you are using this "non-military non-spy" variant of the MIT license?

**Comment by pnkfelix** [comment]
@oli-obk unfortunately I suspect the conditions that you are imposing in your license may make it incompatible with other projects that I would prefer to be able to support, just in the sense of keeping the software itself as free as possible.

I'm inferring that you found this license somewhere.  I'll do some googling and see if I can the original source (perhaps MIT it sounds like?) and also see whether https://opensource.org/licenses or http://choosealicense.com/ have anything to say about it?

**Comment by pnkfelix** [comment]
@oli-obk for example, see the discussion here:

  http://programmers.stackexchange.com/questions/199055/open-source-licenses-that-explicitly-prohibit-military-applications

the people there seem to think that a license like this "non-military non-spy" thing that you have used actually fails the definition of open-source as defined by the OSI and FSF.

**Comment by oli-obk** [comment]
> are you asking if you can just copy over the licenses as is into your crate?

this I'm asking

**Comment by oli-obk** [comment]
> I'm inferring that you found this license somewhere. 

I actually created it just to see if someone would ever notice ;)

**Comment by pnkfelix** [comment]
> > are you asking if you can just copy over the licenses as is into your crate?
> 
> this I'm asking

Oh okay, great!

I was worried you would be firmly committed to the non-military form of your license. (As you can tell from http://programmers.stackexchange.com/questions/199055/open-source-licenses-that-explicitly-prohibit-military-applications , you are not the first person to think of such a thing.)

Not only should there be no problem with you just cut-and-pasting my licenses into your repository, I highly recommend that you actually do so. (And also change the license in a similar manner in the other crates where you have used this, such as `serde-xml`)

I'm not exactly what sure what the white-list filtering _is_ that cargo claims it does (see http://doc.crates.io/manifest.html#package-metadata ), but I strongly suspect that embedding a side-condition like this into a license can be dangerous for other projects that naively import your crate thinking that it is the normal MIT license (even though it is clear from the first line of your license text that something strange is going on).

**Comment by pnkfelix** [comment]
@oli-obk in case you are curious, and since you said you had no clue about licenses: unlike if I had used, say, the GPL, if you had said "I am committed to the non-military clause in my license, I am not going to change it", then I believe I would _not be able_ to stop you from cut-and-pasting the code out of my files and into yours.

The MIT/Apache licenses basically give away the code without any real limits, so that people can incorporate such code into other projects with _stricter_ licenses.

You can compare this to licenses like the GPL, where someone is legally obligated to only copy such code into other projects that are _also_ using a GPL-compatible license. This is why the GPL code and/or its license is sometimes called "viral"

**Comment by oli-obk** [comment]
> unlike if I had used, say, the GPL, if you had said "I am committed to the non-military clause in my license, I am not going to change it", then I believe I would not be able to stop you from cut-and-pasting the code out of my files and into yours.

Licenses are weird, but fear not, I'm all for making sure everyone can use my code.
